geopy/geocoders/googleplaces.py

```diff
@@ -0,0 +1,217 @@
+
+        :param string api_key: The API key required by Google to perform
+            geocoding requests. API keys are managed through the Google APIs
+            console (https://code.google.com/apis/console).
+
+        :param string domain: Should be the localized Google Maps domain to
+            connect to. The default is 'maps.googleapis.com', but if you're
+            geocoding address in the UK (for example), you may want to set it
+            to 'maps.google.co.uk' to properly bias results.
+
+        :param string scheme: Use 'https' or 'http' as the API URL's scheme.
+            Default is https. Note that SSL connections' certificates are not
+            verified.
+
+        :param dict proxies: If specified, routes this geocoder's requests
+            through the specified proxy. E.g., {"https": "192.0.2.0"}. For
+            more information, see documentation on
+            :class:`urllib2.ProxyHandler`.
+        """
+        super(GooglePlaces, self).__init__(
+            scheme=scheme, timeout=timeout, proxies=proxies, user_agent=user_agent
+        )
+
+        self.api_key = api_key
+        self.domain = domain.strip('/')
+        self.scheme = scheme
+        self.doc = {}
+
+        self.autocomplete_api = '%s://%s/maps/api/place/autocomplete/json' % (self.scheme, self.domain)
+        self.details_api = '%s://%s/maps/api/place/details/json' % (self.scheme, self.domain)
+
+    @staticmethod
+    def _format_components_param(components):
+        """
+        Format the components dict to something Google understands.
+        """
+        return "|".join(
+            (":".join(item)
+             for item in components.items()
+             )
+        )
+
+    def geocode(
+            self,
+            query,
+            exactly_one=False,
+            timeout=None,
+            offset=None,
+            location=None,
+            radius=None,
+            language=None,
+            types=None,
+            components=None,
+    ):  # pylint: disable=W0221,R0913
+        """
+        Geocode a location query.
+
+        :param string query: The address or query you wish to geocode.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+            .. versionadded:: 1.12
+
+        :param offset: The position, in the input term, of the last character that the service uses to match predictions.
+            For example, if the input is 'Google' and the offset is 3, the service will match on 'Goo'.
+            If no offset is supplied, the service will use the whole term.
+        :type offset: int
```

Comment:
do we need to expose this in the api here? can't we move this work on to the called? (i just don't see a good usecase for this when calling the api from the backend)